x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/register_transaction_duration_rule_type.ts

```diff
@@ -140,26 +143,33 @@ export function registerTransactionDurationRuleType({
                 ...getDocumentTypeFilterForTransactions(
                   searchAggregatedTransactions
                 ),
-                { term: { [SERVICE_NAME]: ruleParams.serviceName } },
-                {
-                  term: {
-                    [TRANSACTION_TYPE]: ruleParams.transactionType,
-                  },
-                },
+                ...termQuery(SERVICE_NAME, ruleParams.serviceName),
+                ...termQuery(TRANSACTION_TYPE, ruleParams.transactionType),
                 ...environmentQuery(ruleParams.environment),
               ] as QueryDslQueryContainer[],
             },
           },
           aggs: {
-            environments: {
-              terms: {
-                field: SERVICE_ENVIRONMENT,
-                missing: ENVIRONMENT_NOT_DEFINED.value,
+            series: {
+              multi_terms: {
+                terms: [
+                  { field: SERVICE_NAME },
+                  {
+                    field: SERVICE_ENVIRONMENT,
+                    missing: ENVIRONMENT_NOT_DEFINED.value,
+                  },
+                  { field: TRANSACTION_TYPE },
+                ],
+                size: 1000,
+                order: { [TRANSACTION_DURATION]: 'desc' as const },
+              },
+              aggs: {
+                ...averageOrPercentileAgg({
+                  aggregationType: ruleParams.aggregationType,
+                  transactionDurationField: field,
+                }),
+                ...getSourceFieldsAgg(),
```

Comment:
using a top hits agg with a lot of buckets is very expensive. if there is a way to only do this for actual alerts that should fire I would recommend it.